DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -2182,4 +1566,196 @@ internal class ContextMenuCommand
         /// </summary>
         public MethodInfo Method { get; set; }
     }
+
+    #region Default Help
+    /// <summary>
+    /// Represents the default help module.
+    /// </summary>
+    public class DefaultHelpModule : ApplicationCommandsModule
+    {
+        public class DefaultHelpAutoCompleteProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
+                var slashCommands = globalCommands.Concat(guildCommands)
+                    .Where(ac => !ac.Name.Equals("help", StringComparison.OrdinalIgnoreCase))
+                    .GroupBy(ac => ac.Name).Select(x => x.First()).Where(ac => ac.Name.StartsWith(context.Options[0].Value.ToString(), StringComparison.OrdinalIgnoreCase));
+                var list = slashCommands.ToList();
+                foreach (var sc in list.Take(25))
+                {
+                    options.Add(new DiscordApplicationCommandAutocompleteChoice(sc.Name, sc.Name.Trim()));
+                }
+                return options.AsEnumerable();
+            }
+        }
+
+        public class DefaultHelpAutoCompleteLevelOneProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
```

Comment:
same comment regarding firing both of these off at the same time (if we figure out it doesn't break rate limiting)